pilot/pkg/xds/endpoints/endpoint_builder.go

```diff
@@ -399,12 +373,14 @@ func (b *EndpointBuilder) BuildLocalityLbEndpointsFromShards(
 			// The mTLS settings may have changed, invalidating the cache endpoint. Rebuild it
 			needToCompute = true
 		}
-		if needToCompute {
+		if needToCompute || noCache {
 			eep = buildEnvoyLbEndpoint(b, ep, mtlsEnabled)
 			if eep == nil {
 				continue
 			}
-			ep.ComputeEnvoyEndpoint(eep)
+			if noCache {
+				ep.ComputeEnvoyEndpoint(eep)
```

Comment:
confused why set it for cds endpoint, previously it is for eds

Reply:
Should be `!noCache`.